lib/traversal.cc

```diff
@@ -141,7 +160,147 @@ unsigned int Traverser::degree_right(Kmer& node)
     return degree;
 }
 
-unsigned int Traverser::degree(Kmer& node)
+unsigned int Traverser::degree(const Kmer& node)
+const
 {
     return degree_right(node) + degree_left(node);
 }
+
+
+
+template<bool direction>
+AssemblerTraverser<direction>::AssemblerTraverser(const Hashtable * ht,
+        Kmer start_kmer,
+        KmerFilterList filters) :
+    Traverser(ht), filters(filters)
+{
+    cursor = start_kmer;
+}
+
+template<>
+Kmer AssemblerTraverser<LEFT>::get_neighbor(Kmer& node,
+        const char symbol)
+{
+    return get_left(node, symbol);
+}
+
+template<>
+Kmer AssemblerTraverser<RIGHT>::get_neighbor(Kmer& node,
+        const char symbol)
+{
+    return get_right(node, symbol);
+}
+
+template<>
+unsigned int AssemblerTraverser<LEFT>::cursor_degree()
+const
+{
+    return degree_left(cursor);
+}
+
+template<>
+unsigned int AssemblerTraverser<RIGHT>::cursor_degree()
+const
+{
+    return degree_right(cursor);
+}
+
+template <>
+std::string AssemblerTraverser<RIGHT>::join_contigs(std::string& contig_a,
+        std::string& contig_b)
+const
+{
+    return contig_a + contig_b.substr(_ksize);
+}
+
+template <>
+std::string AssemblerTraverser<LEFT>::join_contigs(std::string& contig_a,
+        std::string& contig_b)
+const
+{
+    return contig_b + contig_a.substr(_ksize);
+}
+
+template<bool direction>
+char AssemblerTraverser<direction>::next_symbol()
+{
+    char * symbol_ptr = alphabets::DNA_SIMPLE;
+    char base = *symbol_ptr; // kill -Wmaybe-uninitialized warning
+    short found = 0;
+    Kmer neighbor;
+    Kmer cursor_next;
+
+    while(*symbol_ptr != '\0') {
+        neighbor = get_neighbor(cursor, *symbol_ptr);
```

Comment:
Yes, that has been annoying ever since. In the old system at least comments that were on stuff that changed got folded away, so I could see if people had done stuff (assuming they'd only ever change it towards the suggestion if they didn't also comment) ... alas the world is not perfect :)